### ISSUE TITLE
[Feature-Request] Option for a project-prefix

### ISSUE THREAD

**Issue by Suven** [funkensturm/ember-local-storage#168]
Hey!

It would be great if one could configure a prefix for all storge-objects/the adapter. Maybe this is something one could set one time in a service inside an instance-initializer?

My use-case is just that, when developing multiple apps at once, the Ressources-Tab in Chrome can get a little messy. 

For anybody else who might face this:
You can get around that problem by configuring different local domains for your different projects via `/etc/hosts` and then configuring that host in your `.ember-cli`.

**Comment by fsmanuel** [comment]
@Suven That is actually a good idea and should be easy to implement.
I'm happy to accept a PR and can help point you to the right places in the code.

I guess the best would be to set the `namespace` via environment.js:

```js
// config/environment.js
module.exports = function() {
  var ENV = {
    'ember-local-storage': {
      namespace: 'my-app'
    }
  }
};
```

We would need a helper `getNamespace` in [storage.js](https://github.com/funkensturm/ember-local-storage/blob/master/addon/helpers/storage.js) where we fetch the config.

Next step would be to identify where we generate keys and add the namespace.

**Comment by csprocket777** [comment]
I'm also needing this, however I'd need a new storage object based on another variable.

For example: I'm making a utility that could consume a user provided object and I'd want to store a localStorage object that pertains to what the user provided. And then be able to retrieve that localStorage object again when the user returns and consumes that same user provided object.

**Comment by fsmanuel** [comment]
@csprocket777 can you open a new issue that we can discuss your question. Please add a real example. Right now I don't fully understand your requirements.

**Comment by csprocket777** [comment]
I have opened a new issue explaining my use case.
https://github.com/funkensturm/ember-local-storage/issues/172

**Comment by vst** [comment]
I have a similar situation where I need to create a namespace for the project as I may end up deploying multiple Ember applications under the same sub-domain. 

Looks like Ember Simple Auth provides [an option](https://github.com/simplabs/ember-simple-auth/issues/88#issuecomment-35642258) in `environment.js` similar to @fsmanuel's idea.

I would love to jump into this, but not any soon yet :/

**Comment by fsmanuel** [comment]
@Suven @csprocket777 @vst I opend #279 to add the namespace option. I would appreciate a review if you are still using the addon and have some time.

**Comment by fsmanuel** [comment]
@Suven @vst @csprocket777 after almost a year I released the feature as version `1.6.0`. Sorry for the late release.

**Comment by vst** [comment]
@fsmanuel Thanks for the hard work and keeping us updated.

I am not quite sure when I will attend this, but I will update here or on a new issue if we see any problems.
